### PR TITLE
FHG-5472 Build and Test Workflow Refactor

### PR DESCRIPTION
**Pull request by hypersolutions**
- Added a build and test template with inputs to centralize the build and test steps
- Updated the main build and test to call the workflow

### CODE REVIEW

File: .github/workflows/build-and-test-template.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      dotnet_version:
+        required: true
+        type: string
+      runs_on:
+        required: false
+        type: string
+        default: ubuntu-22.04
+
+jobs:
+  notification-api:
```
Comment (Zac-AND-Digital) [inline .github/workflows/build-and-test-template.yml:21]: Can we change this job name :)

**Comment by Zac-AND-Digital** [comment]
The job names are a little much, do you think you could reduce them to be something like "Build & Test Notification API" or similar?

Screenshot 2024-07-12 at 12 14 34 PM

File: .github/workflows/build-and-test-template.yml (hunk 1 above)
Comment (hypersolutions) [reply .github/workflows/build-and-test-template.yml:21]: Oh yeah, that is wrong 👍

**Review by Zac-AND-Digital — APPROVED** [review]
Great work!

**Review by AaronJY — APPROVED** [review]

**Pull request merged**